File: android/xray/src/main/java/com/applicaster/xray/core/Event.kt
package com.applicaster.xray.core

data class Event(
    val category: String,
    val subsystem: String,
    val timestamp: Long, // UTC
    val level: Int,
    val message: String,
    val data: Map<String, Any>?,
    val context: Map<String, Any>?,
    val exception: Throwable? // todo: not sure about this one, maybe handle it in event builder and add to data?
)

File: apple/xray/xray/Core/Event.swift
//
//  Event.swift
//  xray
//
//  Created by Anton Kononenko on 7/9/20.
//  Copyright © 2020 Applicaster. All rights reserved.
//

import Foundation
import os.log
struct Event {
    //iOS-Category
    let category: String
    let subsystem: String // Substem
    let timestamp: Double = round(NSDate().timeIntervalSince1970) // UTC
    let level: Int
    let message: String
    let data: [String: Any]?  // What func, line
    let context: [String: Any]?
//    let exception: NSException? // todo: not sure about this one, maybe handle it in event builder and add to data?
}
